### PR TITLE
Reducing unit test time

### PR DESCRIPTION
**Pull request by dkalinowski**
### 🛠 Summary
Addressing JIRA tickets:
CVS-150532
CVS-150533

- Shortened waiting time to wait for free infer request from the queue in unit test
- Reduced multi-thread infer requests queue unit test workload to last 0.5s instead of 4 seconds
- Re-use already loaded LLM model wherever possible. Grouped all LLM unit tests with the same server configuration into single test suite.
- Setting max tokens=1 in all unit tests to avoid unlimited llm text generation
- Changed underlying file system poll wait seconds variable from seconds to milliseconds to be able to reduce waiting times (public APIs remain **unchanged** - seconds)
- Changed underlying resource cleaner interval seconds variable from seconds to milliseconds to be able to reduce waiting times (public APIs remain **unchanged** - seconds)
- Tests which wait for config.json to be reloaded or for resources to be cleaned have shortened interval from 1 second to 20s using the changes above
- Moved creation of ContinuousBatchingPipeline out of createLLMNodeResources to separate virtual method (LLMNodeResources::initializeContinuousBatchingPipeline) to be able to mock it and do not load model in tests where it is unnecessary
- Cleanup unused/duplicated code in unit tests (CreateConfig instead of using existing util createConfigFileWithContent)

There are multiple unit tests with over 1.5gb string preparation that take 4s+, those will be removed from unit tests once similar tests appear in functional test suite. It is planned by validation team @pgladkows CVS-151232

Unit tests sorted by execution time before change:
```
LLMJinjaChatTemplateHttpTest.inferChatCompletionsUnary: 5865 ms
TextStreamerTest.noValueReturnedStringWithoutNewLineOrSpace: 4985 ms
LLMJinjaChatTemplateHttpTest.inferCompletionsStream: 4964 ms
LLMJinjaChatTemplateHttpTest.inferCompletionsUnary: 4925 ms
LLMDefaultChatTemplateHttpTest.inferDefaultChatCompletionsUnary: 4914 ms
TextStreamerTest.putReturnsValue: 4880 ms
TextStreamerTest.putDoesNotReturnValueUntilNewLineDetected: 4870 ms
TextStreamerTest.putReturnsValueTextWithSpaces: 4868 ms
TextStreamerTest.valueReturnedCacheCleared: 4859 ms
TextStreamerTest.putReturnsValueAfterEndCalled: 4857 ms
TextStreamerTest.putReturnsValueTextWithNewLineInTheMiddle: 4847 ms
LLMJinjaChatTemplateHttpTest.inferChatCompletionsStream: 4813 ms
LLMOptionsHttpTest.LLMNodeOptionsCheckDefault: 4679 ms
PredictValidationStringNativeTest/0.positive_over_1gb: 4540 ms
PredictValidationStringNativeTest/1.positive_over_1gb: 4352 ms 
EnsembleFlowCustomNodePipelineExecutionTest.MultipleDeinitializeCallsOnRetire: 3604 ms
EnsembleFlowCustomNodePipelineExecutionTest.ReloadPipelineWithoutNodeDeinitializeAllCustomNodes: 3604 ms
ModelManagerCleaner.ConfigReloadShouldCleanupResources: 3603 ms
OVInferRequestQueue.MultiThread: 3040 ms
CAPIStateIntegration.Config: 2172 ms
PredictValidationString1DTest/1.positive_over_1gb: 1695 ms
PredictValidationString1DTest/0.positive_over_1gb: 1657 ms
PredictValidationString2DTest/1.negative_over_1gb_after_expansion: 1636 ms
PredictValidationString2DTest/0.negative_over_1gb_after_expansion: 1634 ms
LLMFlowHttpTest.streamChatCompletionsFinishReasonStop: 1617 ms
LLMHttpParametersValidationTest.MessagesWithOnlyContent: 1172 ms
LLMOptionsHttpTest.LLMNodeOptionsCheckHalfDefault: 1050 ms
LLMConfigHttpTest.LLMNodeNameExists: 1039 ms
LLMOptionsHttpTest.LLMNodeOptionsCheckNonDefault: 1019 ms
ModelManager.ConfigReloadingShouldRetireModelInstancesOfModelRemovedFromJsonRelative: 1017 ms
ModelManager.ConfigReloadingShouldRetireModelInstancesOfModelRemovedFromJson: 1016 ms
OVInferRequestQueue.FullQueue: 1013 ms
TestThreadSafeQueue.NoElementsPushed: 1000 ms
LLMFlowHttpTest.unaryCompletionsJson: 822 ms
...
```

Unit tests sorted by execution time after change:
```
PredictValidationStringNativeTest/0.positive_over_1gb: 4489 ms // will be moved to functional CVS-151232
PredictValidationStringNativeTest/1.positive_over_1gb: 4447 ms // will be moved to functional CVS-151232
PredictValidationString1DTest/0.positive_over_1gb: 1710 ms // will be moved to functional CVS-151232
PredictValidationString1DTest/1.positive_over_1gb: 1676 ms // will be moved to functional CVS-151232
PredictValidationString2DTest/1.negative_over_1gb_after_expansion: 1660 ms // will be moved to functional CVS-151232
PredictValidationString2DTest/0.negative_over_1gb_after_expansion: 1653 ms // will be moved to functional CVS-151232
LLMFlowHttpTest.unaryCompletionsJson: 985 ms
...
```

### 🧪 Checklist

- [x] Unit tests added.
- [x] The documentation updated.
- [x] Change follows security best practices.

### CODE REVIEW

File: src/llm/llmnoderesources.cpp [hunk 1]
```diff
@@ -163,7 +162,11 @@ Status LLMNodeResources::createLLMNodeResources(std::shared_ptr<LLMNodeResources
 
     try {
         plugin_config_t tokenizerPluginConfig = {{"PERFORMANCE_HINT", "THROUGHPUT"}};
-        nodeResources->cbPipe = std::make_unique<ov::genai::ContinuousBatchingPipeline>(basePath, nodeResources->schedulerConfig, nodeResources->device, nodeResources->pluginConfig, tokenizerPluginConfig);
+        auto status = nodeResources->initializeContinuousBatchingPipeline(basePath, nodeResources->schedulerConfig, nodeResources->device, nodeResources->pluginConfig, tokenizerPluginConfig);
+        if (!status.ok()) {
+            SPDLOG_ERROR("Error during llm node initialization for models_path: {}", basePath);
+            return StatusCode::LLM_NODE_RESOURCE_STATE_INITIALIZATION_FAILED;
+        }
```
Comment (mzegla) [inline src/llm/llmnoderesources.cpp:169]: This is unreachable

File: src/llm/llmnoderesources.cpp [hunk 2]
```diff
@@ -182,6 +185,16 @@ Status LLMNodeResources::createLLMNodeResources(std::shared_ptr<LLMNodeResources
     return StatusCode::OK;
 }
 
+Status LLMNodeResources::initializeContinuousBatchingPipeline(
```
Comment (mzegla) [inline src/llm/llmnoderesources.cpp:188]: If we can't return anything else than OK this could be void

File: src/llm/llmnoderesources.hpp [hunk 3]
```diff
@@ -37,6 +37,7 @@
 
 #include "../logging.hpp"
 #include "../stringutils.hpp"
+#include "llm_executor.hpp"
```
Comment (mzegla) [inline src/llm/llmnoderesources.hpp:40]: Why is it needed here now?

File: src/mediapipe_internal/mediapipegraphdefinition.cpp [hunk 4]
```diff
@@ -461,7 +461,7 @@ Status MediapipeGraphDefinition::initializeNodes() {
                 return StatusCode::LLM_NODE_NAME_ALREADY_EXISTS;
             }
 
-            std::shared_ptr<LLMNodeResources> nodeResources = nullptr;
+            std::shared_ptr<LLMNodeResources> nodeResources = std::make_shared<LLMNodeResources>();
```
Comment (mzegla) [inline src/mediapipe_internal/mediapipegraphdefinition.cpp:464]: Why are you making that change?

Comment (dkalinowski) [reply src/mediapipe_internal/mediapipegraphdefinition.cpp:464]: The createLLMNodeResources caller needs to be able to initialize empty resources object on its own, to be able to select which class it should be. In core OVMS it is LLMNodeResources, but in unit tests where loading model is not important, it is MockedLLMNodeResources. There is no possibility to mock when you hide creation inside createLLMNodeResources.

File: src/llm/llmnoderesources.cpp (hunk 1 above)
Comment (dkalinowski) [reply src/llm/llmnoderesources.cpp]: That's right, removed

File: src/llm/llmnoderesources.cpp (hunk 2 above)
Comment (dkalinowski) [reply src/llm/llmnoderesources.cpp]: Right, changed to void

File: src/llm/llmnoderesources.hpp (hunk 3 above)
Comment (dkalinowski) [reply src/llm/llmnoderesources.hpp]: You are right, it is not needed, we can forward declare. After doing so I could even decrease number of expected errors in ci/cpp_clean.sh

**Review by dtrawins — APPROVED** [review]

**Pull request merged**